lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1097,12 +930,19 @@ def reset_stat(tarinfo):
                         continue
 
                 # Dealing with a normal file, just add it by name.
-                tar_file.add(os.path.realpath(b_src_path), arcname=filename, recursive=False, filter=reset_stat)
+                tar_file.add(
+                    to_native(os.path.realpath(b_src_path)),
+                    arcname=filename,
+                    recursive=False,
+                    filter=reset_stat,
+                )
 
-        shutil.copy(b_tar_filepath, b_tar_path)
+        shutil.copy(to_native(b_tar_filepath), to_native(b_tar_path))
         collection_name = "%s.%s" % (collection_manifest['collection_info']['namespace'],
                                      collection_manifest['collection_info']['name'])
+        tar_path = to_text(b_tar_path)
         display.display('Created collection for %s at %s' % (collection_name, to_text(b_tar_path)))
```

Comment:
may as well change this to tar_path now that you have converted it above

```suggestion
        display.display(u'Created collection for %s at %s' % (collection_name, tar_path))
```